### DISCUSSION TITLE
Custom domain issues - configured and stays stuck on hello!

### DISCUSSION THREAD

**Discussion by jxs714** [pretix/pretix#3645] [Q&A] [answered]
Hey guys,

I am having a hard time with custom domain configuration on self hosted. I am using Nginx and i've setup the following in nginx.

DOMAIN1 is the domain I am using pretix on
DOMAIN2 is the domain I am trying to mask as an organizer / event

This is setup on DOMAIN2

location / {
	proxy_pass https://tickets.DOMAIN1.com;
	proxy_set_header Host tickets.DOMAIN1.com;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_redirect off;
	proxy_set_header Accept-Encoding "";
	sub_filter 'https://tickets.DOMAIN1.com' '/';
	sub_filter_once off;
	proxy_set_header Content-Security-Policy "upgrade-insecure-requests";
	proxy_ssl_server_name on;
	proxy_ssl_name $proxy_host;
}

location /aw {
	proxy_pass https://tickets.DOMAIN1.com;
	proxy_set_header Host tickets.DOMAIN1.com;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_redirect off;
	proxy_set_header Accept-Encoding "";
	sub_filter 'https://tickets.DOMAIN1.com' '/';
	sub_filter_once off;
	proxy_set_header Content-Security-Policy "upgrade-insecure-requests";
	proxy_ssl_server_name on;
	proxy_ssl_name $proxy_host;
}

------

Two ways ive tried customization:

**1.  Without setting up custom domain in Pretix settings**
When I go into pretix and leave the custom domain settings for both organizer and event to default without custom domain, masking works to mask the DOMAIN1 with DOMAIN2, but when checking out to buy a ticket, I get "We could not verify that this request really was sent from you. For security reasons, we therefore cannot process it.

Please go back to the last page, refresh this page and then try again. If the problem persists, please get in touch with us" - I cannot go any further than this

2. Setting up custom domain in pretix
When I go into the organizer area and enter tickets.DOMAIN2.com with the short form being "aw", it doesnt work. I can visit the page at tickets.DOMAIN2.com/aw but it stays stuck on "The selected organizer was not found". I deleted this setting in organizer page and tried the same thing with custom domain instead for event page and it loads up the ticket page with tickets.DOMAIN2.com/aw, but when I add ticket to cart it sends me to the "hello!" home page. 

Any insight on how I can fix this?

**Comment by raphaelm** [comment, marked as answer]
```
 proxy_set_header Host tickets.DOMAIN1.com;
```

This forces pretix to only always see that header. You can compare with our docs: https://docs.pretix.eu/en/latest/admin/installation/manual_smallscale.html#ssl

**Reply by jxs714** [reply]
Hi Raphael,

Thanks for the update and response. I entered this in nginx for tickets.domain1.com based on the example on page and nothing changes. If I use custom domain URL for an event, the page loads and shows tickets for an event. When I click to get tickets, it goes back to the "Hello" root page. 

tickets.DOMAIN1.com NGINX settings:

location / {
	proxy_pass http://localhost:8345;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto https;
	proxy_set_header Host $http_host;
	return 301 https://$host$request_uri;
}

No matter what I do in the pretix dashboard to enter the custom domain to tickets.DOMAIN2.com, it only goes to the root domain. tickets.DOMAIN2.com/aw (organizer name) just goes back to the root. If I change the event custom domain instead, it shows the event but does nothing but go to root when clicking for tickets.

**Comment by jxs714** [comment]
Hi Raphael,

Looks like ive finally fixed it after a whole 2-3 days of scavenge hunting on google search and getting the correct answer from you lol. I really appreciate your help. The nginx settings needed to be set up on tickets.DOMAIN2.com, not on tickets.DOMAIN1.com. 

Thank you so much and keep up the great work
